Source/automap.cpp

```diff
@@ -289,241 +307,190 @@ void __cdecl DrawAutomap()
-		ENG_set_pixel(v4, AutoMapYPos + v5, 200);
-		v3 = automap_type;
+	int do_right;
+	int do_left;
+	int do_cave_left;
+	int do_cave_right;
+	int x1, y1, x2, y2;
+
+	UCHAR flags = automap_type >> 8;
+
+	if ( flags & MAPFLAG_SOLID ) {
+		ENG_set_pixel(screen_x, screen_y, COLOR_DIM);
+		ENG_set_pixel(screen_x - AMPlayerX, screen_y - AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x - AMPlayerX, screen_y + AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x + AMPlayerX, screen_y - AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x + AMPlayerX, screen_y + AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x - AutoMapYPos, screen_y, COLOR_DIM);
+		ENG_set_pixel(screen_x + AutoMapYPos, screen_y, COLOR_DIM);
+		ENG_set_pixel(screen_x, screen_y - AMPlayerX, COLOR_DIM);
+		ENG_set_pixel(screen_x, screen_y + AMPlayerX, COLOR_DIM);
+		ENG_set_pixel(screen_x + AMPlayerX - AutoMapXPos, screen_y + AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x - AMPlayerX + AutoMapXPos, screen_y + AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x - AutoMapYPos, screen_y + AMPlayerX, COLOR_DIM);
+		ENG_set_pixel(screen_x + AutoMapYPos, screen_y + AMPlayerX, COLOR_DIM);
+		ENG_set_pixel(screen_x - AMPlayerX, screen_y + AutoMapYPos - AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x + AMPlayerX, screen_y + AutoMapYPos - AMPlayerY, COLOR_DIM);
+		ENG_set_pixel(screen_x, screen_y + AutoMapYPos, COLOR_DIM);
 	}
-	if ( automap_type < 0 )
-	{
-		DrawLine(v4 - AMPlayerX, v5 - AMPlayerX - AMPlayerY, v4 + AMPlayerX + AutoMapYPos, AMPlayerY + v5, 144);
-		DrawLine(v4 - AutoMapYPos, v5 - AMPlayerX, AutoMapYPos + v4, AMPlayerX + v5, 144);
-		DrawLine(v4 - AutoMapYPos - AMPlayerX, v5 - AMPlayerY, AMPlayerX + v4, v5 + AMPlayerX + AMPlayerY, 144);
-		DrawLine(v4 - AutoMapXPos, v5, v4, v5 + AutoMapYPos, 144);
-		v3 = automap_type;
+
+	if ( flags & MAPFLAG_STAIRS ) {
+		DrawLine(screen_x - AMPlayerX, screen_y - AMPlayerX - AMPlayerY, screen_x + AMPlayerX + AutoMapYPos, screen_y + AMPlayerY, COLOR_BRIGHT);
+		DrawLine(screen_x - AutoMapYPos, screen_y - AMPlayerX, screen_x + AutoMapYPos, screen_y + AMPlayerX, COLOR_BRIGHT);
+		DrawLine(screen_x - AutoMapYPos - AMPlayerX, screen_y - AMPlayerY, screen_x + AMPlayerX, screen_y + AMPlayerX + AMPlayerY, COLOR_BRIGHT);
+		DrawLine(screen_x - AutoMapXPos, screen_y, screen_x, screen_y + AutoMapYPos, COLOR_BRIGHT);
 	}
-	v31 = 0;
-	v30 = 0;
-	v32 = 0;
-	switch ( v3 & 0xF )
-	{
-		case 1:
-			a3 = v4 - AutoMapYPos + AutoMapXPos;
-			a1 = v4 - AutoMapYPos;
-			a2 = v5 - AutoMapYPos;
-			automap_typea = v5 - AMPlayerX;
-			DrawLine(v4, v5 - AutoMapYPos, v4 - AutoMapYPos, v5 - AMPlayerX, 200);
-			DrawLine(v4, a2, a3, automap_typea, 200);
-			DrawLine(v4, v5, a1, automap_typea, 200);
-			v9 = v5;
-			v29 = automap_typea;
-			v28 = a3;
-			goto LABEL_36;
+
+	do_right = 0;
+	do_left = 0;
+	do_cave_right = 0;
+	do_cave_left = 0;
+	switch ( automap_type & MAPFLAG_TYPE ) {
+		case 1: // stand-alone column or other unpassable object

```

Comment:
Use enum for the case comparands.

I.e. `MAPFLAG_DOORR` instead of `1` and `MAPFLAG_OBJECTR | MAPFLAG_DOORR` instead of `5`.

Reply:
It's not a bitmask, they're distinct values. I don't have enough confidence in their meaning to name them.

Reply:
Sure, lets leave them for now and see if intuition arises further along.